web/components/App.jsx

```diff
@@ -0,0 +1,215 @@
+const bdk = require('@salesforce/refocus-bdk')(config, botName);
+import MessageService from '../services/MessageService';
+import AttachmentService from '../services/AttachmentService';
+
+const ZERO = 0;
+const DFB = 5; // Distance From Bottom
+const messageService = new MessageService(bdk);
+const attachmentService = new AttachmentService(bdk);
+
+/**
+ * @param {object} props
+ * @returns {JSX} app container.
+ */
+export default function App(props) {
+  let container;
+  const { response } = props;
+  const [toast, setToast] = useState(false);
+  const [events, setEvents] = useState(response);
+  const [currentText, setCurrentText] = useState('');
+  const [scroll, setScroll] = useState(true);
+  const [fileDraggedOver, setFileDraggedOver] = useState(false);
+  const [filter, setFilter] = useState('CommentAttachment');
+  const [pendingMessage, setPendingMessage] = useState(false);
+
+  /**
+   * @param {array} previousEvents
+   * @param {array} newEvents
+   */
+  function handleNewEventsReceived(previousEvents, newEvents) {
+    const allEvents = previousEvents.concat(newEvents).filter((value, index, self) => {
+      const duplicates = _.filter(self.slice(ZERO, index), ['id', value.id]);
+      return duplicates.length === ZERO ? value : false;
+    }).sort((a, b) => moment(a.createdAt).diff(moment(b.createdAt)));
+
+    setEvents(allEvents);
+
+    if (container && container.scrollHeight - container.scrollTop - DFB <= container.clientHeight) {
+      setScroll(true);
+    } else {
+      const latestEvent = newEvents.length && newEvents[ZERO];
+      setToast(latestEvent && latestEvent.context &&
+        (filter === 'All' || filter.includes(latestEvent.context.type)));
+    }
+  }
+
+  useEffect(() => {
+    const previousEvents = events;
+    const latestEvent = response.length && response[ZERO];
+
+    if (scroll && container) {
+      container.scrollTop = container.scrollHeight;
+      setScroll(false);
+    }
+
+    if (latestEvent && !_.find(previousEvents, latestEvent)) {
+      handleNewEventsReceived(previousEvents, response);
+    }
+  });
+
+  /**
+   * changes the active message filter
+   * @param {object} type - filter type that was selected
+   */
+  function changeFilterType(type) {
+    if (filter === 'All' || type === 'All') {
+      setFilter(type);
+    } else if (filter === type) {
+      setFilter('All');
+    } else if (filter.includes(type)) {
+      setFilter(filter.replace(type, ''));
+    } else {
+      setFilter(filter + type);
+    }
+
+    props.getEventsByType(type);
+    setScroll(true);
+  }
+
+  /**
+   * sends a chat message.
+   */
+  async function sendChat() {
+    if (currentText === '') return;
+    setPendingMessage(true);
+
+    try {
+      await messageService.sendMessage(currentText, props.user);
+      setCurrentText('');
+      setPendingMessage(false);
+    } catch (err){
+      console.error(`message failed to send - ${err.message}`);
```

Comment:
Is this caught elsewhere so an error is displayed to the user? Or is it only console logged

Reply:
Handling errors for message sending wasn't a feature of Timeline-Bot before. It might be a new story in our backlog so the scope of this doesn't grow too much

Reply:
yeah id say its fine like this in terms of the scope of this story